homeassistant/components/nws/sensor.py

```diff
@@ -0,0 +1,149 @@
+"""Sensors for National Weather Service (NWS)."""
+from homeassistant.components.sensor import SensorEntity
+from homeassistant.const import (
+    ATTR_ATTRIBUTION,
+    ATTR_DEVICE_CLASS,
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    LENGTH_KILOMETERS,
+    LENGTH_METERS,
+    LENGTH_MILES,
+    PRESSURE_INHG,
+    PRESSURE_PA,
+    SPEED_MILES_PER_HOUR,
+)
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+from homeassistant.util.distance import convert as convert_distance
+from homeassistant.util.dt import utcnow
+from homeassistant.util.pressure import convert as convert_pressure
+
+from . import base_unique_id
+from .const import (
+    ATTR_ICON,
+    ATTR_LABEL,
+    ATTR_UNIT,
+    ATTR_UNIT_CONVERT,
+    ATTRIBUTION,
+    CONF_STATION,
+    COORDINATOR_OBSERVATION,
+    DOMAIN,
+    NWS_DATA,
+    OBSERVATION_VALID_TIME,
+    SENSOR_TYPES,
+)
+
+PARALLEL_UPDATES = 0
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the NWS weather platform."""
+    hass_data = hass.data[DOMAIN][entry.entry_id]
+    station = entry.data[CONF_STATION]
+    for sensor_type, sensor_data in SENSOR_TYPES.items():
+        if hass.config.units.is_metric:
+            unit = sensor_data[ATTR_UNIT]
+        else:
+            unit = sensor_data[ATTR_UNIT_CONVERT]
+        async_add_entities(
```

Comment:
Please collect all entities first and then make a single call to `async_add_entities`.